package-res/resources/web/pentaho/data/Filter.js

```diff
@@ -22,8 +22,22 @@ define([
 ], function(IsEqual, IsIn, AndFilter, OrFilter, NotFilter) {
   "use strict";
 
-
-  var RootFilter = OrFilter.extend({
+  /**
+   * @name RootFilter
+   * @memberOf pentaho.data.filter
+   * @class
+   * @ignore
+   *
+   * @classdesc The `RootFilter` class implements a type of
+   * `AbstractTreeFilter` {@link pentaho.data.filter.AbstractTreeFilter}. Specifically
+   * the `RootFilter` extends the `RootFilter` {@link pentaho.data.filter.Or} filter.
+   *
+   * @description Creates a `RootFilter` filter that performs the union of a list of given `AbstractPropertyFilter`'s {@link pentaho.data.filter.AbstractPropertyFilter}.
+   *
```

Comment:
It performs a union of descendants of `AbstractFilter`', not just the leaf-like descendants of `AbstractPropertyFilter`.

Also, no need to use

> `AbstractPropertyFilter`'s {@link pentaho.data.filter.AbstractPropertyFilter}

Instead, use  merely 

> {@link pentaho.data.filter.AbstractPropertyFilter}

or 

> [ `AbstractPropertyFilter`'s]{@link pentaho.data.filter.AbstractPropertyFilter}